packages/medusa/src/services/shipping-option.ts

```diff
@@ -397,6 +398,42 @@ class ShippingOptionService extends TransactionBaseService {
     return option
   }
 
+  private async validateAndMutatePrice(
+    option: ShippingOption | CreateShippingOptionInput,
+    priceInput: ValidatePriceTypeAndAmountInput
+  ): Promise<Omit<ShippingOption, "beforeInsert"> | CreateShippingOptionInput> {
+    const option_:
+      | Omit<ShippingOption, "beforeInsert">
+      | CreateShippingOptionInput = { ...option }
+
+    if (isDefined(priceInput.amount)) {
+      option_.amount = priceInput.amount
+    }
+
+    if (isDefined(priceInput.price_type)) {
+      option_.price_type = await this.validatePriceType_(
+        priceInput.price_type,
+        option_ as ShippingOption
+      )
+
+      if (priceInput.price_type === ShippingOptionPriceType.CALCULATED) {
+        option_.amount = null
+      }
+    }
+
+    if (
+      option_.price_type === ShippingOptionPriceType.FLAT_RATE &&
+      (option_.amount == null || option_.amount < 0)
```

Comment:
@adrien2p found a minor bug when using `!option_?.amount` as this is true when `amount = 0` (whish is allowed). 

I've added `==` since this checks both for `undefined` and `null` but not `0`. 

Let me know if you are still fine with the changes.

Reply:
indeed ) is considered false as well. good catch.
The check looks fine to me 🚀